clustering/cluster_minifigures.py

```diff
@@ -0,0 +1,34 @@
+def list_categories(file):
+    with open(file) as minifigures:
+        categories_list = []
+        for line in minifigures:
+            if 'Category ID' in line:
+                continue
+            content = line.strip().split('\t')
+            if len(content) < 2:
+                continue
+            category_id = content[0]
+            if category_id not in categories_list:
+                categories_list.append(category_id)
+    return categories_list
+
+categories = list_categories('minifigures.txt')
+
+with open('categories.json', 'w') as json_file:
+    for element in categories:
+        minifigures_list = []
```

Comment:
move this and the line `json_file.write('{\'category_id\': ' + str(element) + ',\'minifigure_ids\': ' + str(minifigures_list) + '}') `outside of the for loop to create a valid json

Reply:
If it was outside of the loop, wouldn't it just return the last category_id and it's minifigure id's?

Reply:
Nevermind my comment. I managed to do as you supposed by using a list for all dict entries and it works! Big thx 
for your feedback